### PR TITLE
Code refactoring on UpdateRefreshTokenAsync

### PR DESCRIPTION
**Pull request by cguldogan**
**What issue does this PR address?**
With this PR I would like to refactor and simplify Update RefreshToken Async method.

**Does this PR introduce a breaking change?**
No

**Please check if the PR fulfills these requirements**
- [x] The commit follows our [guidelines](https://github.com/IdentityServer/IdentityServer4/blob/dev/.github/CONTRIBUTING.md)
- [x] Unit Tests for the changes have been added (for bug fixes / features)

**Other information**:

### CODE REVIEW

File: src/IdentityServer4/Services/DefaultRefreshTokenService.cs [hunk 1]
```diff
@@ -99,55 +99,71 @@ public DefaultRefreshTokenService(ISystemClock clock, IRefreshTokenStore refresh
...
-                }
-
-                refreshToken.Lifetime = newLifetime;
-                needsUpdate = true;
-            }
-
-            if (needsCreate)
-            {
+                needsCreate = await RemoveRefreshToken(handle);
                 handle = await RefreshTokenStore.StoreRefreshTokenAsync(refreshToken);
```
Comment (veikkoeeva) [inline src/IdentityServer4/Services/DefaultRefreshTokenService.cs]: Should all of these `async`  tasks also have `.ConfigureAwait(false)`? It doesn't appear the codebase has much them, but maybe it's OK to add them as one refactor anyhow. This doesn't only help with "GUI systems", but also with systems like https://github.com/dotnet/orleans (maybe even ASP.NET Core).

**Comment by leastprivilege** [comment]
Thanks. But since you changed an interface . this is a breaking change and has to wait til 3.0

**Comment by cguldogan** [comment]
Hi, @leastprivilege, what shall do I now ?

File: src/IdentityServer4/Services/DefaultRefreshTokenService.cs (hunk 1 above)
Comment (cornem) [reply src/IdentityServer4/Services/DefaultRefreshTokenService.cs]: `ConfigureAwait(false)` is not necessary in ASP.NET Core, as it [doesn't have a synchronization context](https://stackoverflow.com/a/42058544/1746803).

**Pull request closed without merging**